runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -994,12 +994,6 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          return cursor;
       }
 
-   reloRecord->setSize(reloTarget, relocation->getSizeOfRelocationData());
-   reloRecord->setType(reloTarget, kind);
-
-   uint8_t wideOffsets = relocation->needsWideOffsets() ? RELOCATION_TYPE_WIDE_OFFSET : 0;
-   reloRecord->setFlag(reloTarget, wideOffsets);
-
```

Comment:
The reason for this is because of an ordering issue, namely:
1. Do platform specific way of gathering data for the relocation header, if it exists
2. Otherwse, do common way of gathering data for the relocation header

If a relo type `TR_ReloType` has both a platform specific implementation and a common implementation, the platform specific implementation must be done. This means that even though this deleted code is common regardless of the relotype, we cannot move it into the code that does the common way of gathering data, because we might not call into it. Therefore, this code is moved into each of the platform specific versions of `initializeAOTRelocationHeader`

Reply:
not a huge fan of spraying code like that. Did you consider defining a "common" initialization routine that must be called for all platforms followed/preceded by a call to either the platform or "default" initialization code? You'd still have the flexibility to move stuff from the "common" initialization routine to all the platform routines if needed, but if you could share it, it can go into that common routine.

Reply:
FWIW, this code was technically redundant because the the `TR_RelocationRecordBinaryTemplate` fields were being initialized in each of the various implementations via the raw pointer manipulation, before executing this code, so I definitely didn't increase the code duplication, I just didn't decrease it heh :).

However, adding a common pre-initialization routine is definitely a good idea; I will add it in the misc clean up PR that I'll open after this. Given how big this PR is I don't want to complicate it further or introduce some stupid bug.